feat(extui): use winborder for msg window

seblyng opened the pull request:
Problem:  The message window is essentially a regular floating window
          but does not use 'winborder'.
          Still some "scratch" buffer options unset after it was removed
          from its window.
Solution: Do not set the border when opening the window message.
          Forego passing `scratch = true` when opening a buffer,
          set the options manually when necessary.

justinmk commented on runtime/lua/vim/_extui/shared.lua:
```diff
@@ -42,12 +42,13 @@ function M.check_targets()
       or not api.nvim_win_is_valid(M.wins[type])
       or not api.nvim_win_get_config(M.wins[type]).zindex -- no longer floating
     then
+      local winborder = vim.o.winborder
```
```suggestion
      local winborder = vim.o.winborder == '' and 'single' or vim.o.winborder
```

seblyng replied:
I can do it, but stylua breaks it up into two lines which I didn't like

justinmk commented:
will wait for comments from @luukvbaal

luukvbaal commented:
Let's just get rid of the "single" default then? Seems less surprising and `border = nil` then uses 'winborder' automatically.

The cmdline, prompt and pager window are floating windows masquerading as regular windows (w.r.t position/dimensions), so I think it makes sense to keep the custom single line top border for those.

seblyng commented:
> The cmdline, prompt and pager window are floating windows masquerading as regular windows (w.r.t position/dimensions), so I think it makes sense to keep the custom single line top border for those.

I agree. I do sort of wish that it explicitly set the background to `Normal` instead of `NormalFloat` for those "floating windows" as it sort of just looks like a split. However, I can also override that with an autocmd on the FileType, so not a huge deal

luukvbaal committed:
fix(extui): use 'winborder' for message window

Problem:  The message window is essentially a regular floating window
          but does not use 'winborder'.
          Still some "scratch" buffer options unset after it was removed
          from its window.
Solution: Do not set the border when opening the window message.
          Forego passing `scratch = true` when opening a buffer,
          set the options manually when necessary.

Co-authored-by: Sebastian Lyng Johansen <sebastian@lyngjohansen.com>

luukvbaal commented:
There is this TODO note:
https://github.com/neovim/neovim/blob/8aed423072a37bfa04263263d45c91263e1072d3/runtime/lua/vim/_extui.lua#L90-L92

The prompt and pager windows are both still neighboring the cmdline so I think it makes sense to use the `MsgArea` group for these. `NormalFloat` seems fine for the message window for now.
